frame/oracle/src/lib.rs

```diff
@@ -502,18 +511,25 @@ pub mod pallet {
 			let who = ensure_signed(origin)?;
 			let author_stake = OracleStake::<T>::get(&who).unwrap_or_else(Zero::zero);
 			ensure!(Self::is_requested(&asset_id), Error::<T>::PriceNotRequested);
-			ensure!(author_stake >= T::MinStake::get(), Error::<T>::NotEnoughStake);
+			ensure!(
+				author_stake >=
+					T::MinStake::get().saturating_add(
+						Self::answer_in_transit(&who).unwrap_or_else(Zero::zero)
+					),
+				Error::<T>::NotEnoughStake
+			);
 
 			let set_price = PrePrice {
 				price,
 				block: frame_system::Pallet::<T>::block_number(),
 				who: who.clone(),
 			};
+			let asset_info = AssetsInfo::<T>::get(asset_id);
```

Comment:
It's probably not a big deal here but I think we should avoid using `ValueQuery` on this kind of storage as the `get` will return the `Default` implementation if I am not doing a mistake? This will be backdoor to bug when the code will evolve? This `ValueQuery` really look like the _null_ of substrate... Please correct me if I am wrong.

Reply:
I tend to do valueQuery by default and I believe parity does as well. If I understand correctly, I could be down to change them if you make a good case and have a potential example, but that would be a lot of code changes, as I would have to unwrap every instance and fix the tests and benchamrks too. So I would say down to do it after but not in scope of this PR

Reply:
Let's take the `submit_price` extrinsic, if an oracle is passing a non existing asset, should the extrinsic be successful? Unless am missed something, it look like it will be successful

Reply:
yes it will fail with max prices as the max is 0 if an asset is not set, but I see your point, will make a PR after this to fix it